### PR TITLE
Retry retrieval of service status, when the response calling health url is not 2xx.

### PR DESCRIPTION
**Pull request by SteKoe**

### CODE REVIEW

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusInfoNotOkException.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	public Map<String, Object> getDetails() {
+		Map<String, Object> details = new LinkedHashMap<>();
+		details.put("status", httpStatus.value());
+		details.put("error", httpStatus.getReasonPhrase());
+		if (body.get("details") instanceof Map) {
+			details.putAll((Map<? extends String, ?>) body.get("details"));
+		}
+		else {
+			details.putAll(body);
+		}
```
Comment (erikpetzold) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusInfoNotOkException.java:45]: this part is strange (I know you just moved it from somewhere else)

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java [hunk 2]
```diff
@@ -74,6 +74,8 @@ public StatusUpdater(InstanceRepository repository, InstanceWebClient instanceWe
 		log.debug("Update status for {}", instance);
 		return this.instanceWebClient.instance(instance).get().uri(Endpoint.HEALTH)
 				.exchangeToMono(this::convertStatusInfo).log(log.getName(), Level.FINEST)
+				.retryWhen(Retry.max(5)
```
Comment (erikpetzold) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java:77]: Can this "hide" real outages/restarts?
-> service is down for a short moment, but due to the retries we don not see this in event log
-> can we differentiate between network errors and service down?

Comment (erikpetzold) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java:77]: should this be configurable?

**Comment by SteKoe** [comment]
There is already a retry mechanism for non-modifying requests in `spring.boot.admin.monitor.default-retries` as documented here: https://codecentric.github.io/spring-boot-admin/2.6.2/#_configuration_options.

Hence, this PR will be closed.

See also #1711

**Pull request closed without merging**